No HI

Pmaene opened the pull request:
Added support to exclude Hearing Impaired subtitles from the results.

michaelbaudino commented on lib/addic7ed/subtitle.rb:
```diff
@@ -18,8 +19,13 @@ def to_s
+      hi_works = true
+      if @hi and no_hi
+        hi_works = false
+      end
```
This whole logic could simply be replaced by `hi_works = !no_hi || !@hi`

michaelbaudino commented on lib/addic7ed/episode.rb:
```diff
@@ -48,10 +48,10 @@ def find_subtitles(lang)
     end
 
-    def find_best_subtitle(lang)
+    def find_best_subtitle(lang, no_hi)
```
Can you make this argument optional like in `best_subtitle` ?

michaelbaudino commented on lib/addic7ed/episode.rb:
```diff
@@ -23,18 +23,18 @@ def subtitles(lang = 'fr')
     end
 
-    def download_best_subtitle!(lang, http_redirect_limit = 8)
+    def download_best_subtitle!(lang, no_hi, http_redirect_limit = 8)
```
Having `no_hi` parameter optional here would be great for initial call.

michaelbaudino commented:
@Pmaene: same comment as on #12 : I'm absolutely confused I missed this PR months ago, but it's again very good work :+1: 
If you can address my comments, I'll be very happy to merge and have this new feature :shipit:

michaelbaudino commented:
Ho, also, it does not merge :disappointed:

michaelbaudino commented:
And would it be a lot of work to add a `--hi-only` option for a11y ?

Pmaene commented:
Here you go 😊. I noticed that my test case was not correct, so I have also updated that one. I added "filename_no_hi", although it would work with the original "filename" with the changes from #12. I chose English instead of French, because English has both kinds of subs (HI and no HI).

I think it would be quite a bit of work to add an --hi-only, as we'd need to refactor everything to support either no preference, only HI or only not HI.

michaelbaudino commented on spec/lib/addic7ed/episode_spec.rb:
```diff
@@ -108,34 +116,34 @@
     it 'gets the best subtitle candidate with a network request' do
       expect(episode).to receive(:best_subtitle).once.and_call_original
-      episode.download_best_subtitle!('fr')
+      episode.download_best_subtitle!('fr', false)
```
The second argument here (`false`) can now be removed :smiley: 
Same thing at lines 126, 133, 139, 146, 156 and 162.

michaelbaudino commented:
Thanks, we're almost there :+1: :smile:
Can you merge `master` branch into this one? (because it has conflicts right now and thus I cannot merge it)

Pmaene commented:
Removed the argument and merged all upstream changes 😃.

michaelbaudino commented:
Thanks a lot :+1:
